db/compaction/compaction_job.cc

```diff
@@ -198,6 +201,21 @@ struct CompactionJob::SubcompactionState {
 
   SubcompactionState& operator=(const SubcompactionState&) = delete;
 
+  // Adds the key and value to the builder
+  // If paranoid is true, adds the key-value to the paranoid hash
+  void AddToBuilder(const Slice& key, const Slice& value, bool paranoid) {
+    auto curr = current_output();
+    assert(builder != nullptr);
+    assert(curr != nullptr);
+    builder->Add(key, value);
```

Comment:
I'm paranoid but can we reorder adding to builder and calculating hash? This would cover memory corruption in key and value itself within builder->add().